examples/main.go

```diff
@@ -35,21 +73,16 @@ func main() {
 		LNURLConfig: ln.LNURLoptions{
 			Address: os.Getenv("LNURL_ADDRESS"),
 		},
-	})
-	if err != nil {
-		log.Fatal(err)
 	}
-	lsatmiddleware, err := ginlsat.NewLsatMiddleware(&ginlsat.GinLsatMiddleware{
-		Amount:   5,
-		LNClient: lnClient,
-	})
+	lsatmiddleware, err := ginlsat.NewLsatMiddleware(lnClientConfig, AmountFunc)
 	if err != nil {
 		log.Fatal(err)
 	}
 
 	router.Use(lsatmiddleware.Handler)
 
 	router.GET("/protected", func(c *gin.Context) {
+		c.Request = FiatToBTC("USD", 0.01)
```

Comment:
You can drop this as well. It doesn't do anything, right?

Reply:
I wrote the function `FiatToBTC` that is providing a request to `AmountFunc` to demonstrate how the amount can be changed dynamically. So, it's creating a request for converting 0.01 USD to Sats by hitting `https://blockchain.info/tobtc?currency=USD&value=0.01` that is executing within `AmountFunc` in this example.

Reply:
What do you think about this: https://github.com/getAlby/gin-lsat/pull/13/commits/f5d45952c4f05a0fcc06c762717ac7addac5df0c , this looks more clean to me. You shouldn't reuse the same request from c.Request just to make a request to fetch the price.